### ISSUE TITLE
Automation: ReferenceError [Error]: x is not defined

### ISSUE THREAD

**Issue by aleksa-krolls** [OpenFn/ConSoSci#164]
## Describe the bug
I re-ran the automation to sync the `TT_SiteRegistration` form with the `wcs test mssql db` to ensure the SQL is executed. 

I am now testing the auto-generated job against data and encountering the error `ReferenceError [Error]: x is not defined`. Can you please edit the `TT_SiteRegistration.js` job and/or advise if edits need to be made to the automation solution? 

## To Reproduce
See run failures: 
https://www.openfn.org/projects/p5qvqx/runs/0618a66f-0a23-74b7-8efa-6217ee834c6f
https://www.openfn.org/projects/p5qvqx/runs/0618a67a-2634-7c42-ad50-494301bf3334

### expression.js
Auto-generated job: https://www.openfn.org/projects/p5qvqx/jobs/jyz8jn

A1 run that triggered this job to be created: https://www.openfn.org/projects/p5qvqx/runs/0618a66b-e998-7a1d-aeaa-83594fa99a7a

### state.json
NOTE: This is running in the `Trillion Trees` openfn project
Kobo: `Trillion Trees Kobo`
DB: `wcs test mssql db`
You can run this job against this state to reproduce: 
https://www.openfn.org/projects/p5qvqx/messages/06189351-971a-72bc-8bad-f310a7a3f765
https://www.openfn.org/projects/p5qvqx/messages/06189355-06c0-754f-b84a-6222e650aefa

**Comment by lakhassane** [comment]
@Aleksa Krolls I solved the x issue. However while running the TT_SiteRegistration.js job I’m having this issue with null value. Are the state complete? Or should some data be inserted first in the DB for lookup tables.
<img width="1425" alt="Screen Shot 2021-11-09 at 17 33 24" src="https://user-images.githubusercontent.com/8411231/140976437-bab2dea1-469a-4fd2-9145-07e720b119e8.png">

FYI this is the query that returns null and there are some others that will do same
  ```
TT_LengthID_FallowPeriod: await findValue({
        uuid: 'tt_lengthid',
        relation: 'TT_length',
        where: { TT_lengthExtCode: dataValue('fallow_period') },
      })(state),
```

**Comment by aleksa-krolls** [comment]
Thank you @lakhassane. I was able to hand back to WCS this morning and they already implemented mapping changes to resolve the issues they're describing above. **Fyi, they will continue to make job mapping changes this morning... so if we make any further edits, let's push to a separate branch so we can merge later.** 

Two questions: 
1. What was causing the `x` issue? Does this require further changes to the automation solution? 
2. I'm seeing something funny going on with the `upsertMany(...)` mappings. Search for `undefined` in this section to see what I mean... did these mappings not generate properly? 
https://github.com/OpenFn/ConSoSci/blob/master/automation/trillion-trees/TT_SiteRegistration.js#L60-L128

Here are some sample states that have data for some of these upsertMany mappings: 
1. `where_animals_shifted`: https://www.openfn.org/projects/p5qvqx/messages/06189355-06c0-754f-b84a-6222e650aefa
2. `type_grazing_animals`: https://www.openfn.org/projects/p5qvqx/messages/06189352-5bc3-75b3-b323-d21d6d47a343

**Comment by lakhassane** [comment]
@aleksa-krolls 
1. yes I had to update the automation solution (`syncToOpenFn.js`) because we had a typo in the depth of some questions. Depth was wrongly set to greater than 1 thus acting as if they were nested which weren't the case.
2. This issue is linked to the above one still for some select_multiple questions. If they are not nested, they do not have a composed path. So, we should find a specific way of generated them as well (changes done to `syncToOpenFn`).

I made a fix for both and pushed on prod.
